### ISSUE TITLE
"TARDIS loading" icon while it's being spawned

### ISSUE THREAD

**Issue by parar020100** [MattJeanes/TARDIS#1066]
Spawning the TARDIS usually freezes the game (especially if it's first time), and there's no indication that it's okay

Suggested by @FuzzyLeo and @NovaAstral in Discord:
> what about adding some form of visual to indicate its loading? 

> jazztronauts has to have game freezes due to the nature of it, but when it does freeze it put a thing on screen to indicate that its just loading

> it would help people understand why theres lag probably, but it would have to show up on every players screen (especially the host if its p2p) as everyone will experience a freeze, so it would need to be unobtrusive 

> a small tardis loading icon in the corner would work i think

[discord discussion](https://discord.com/channels/1192662071817535598/1195242754339508254)

**Comment by FuzzyLeo** [comment]
Jazztronauts does this so it would be worth looking into how they do it

**Comment by NovaAstral** [comment]
> Jazztronauts does this so it would be worth looking into how they do it

It's likely they just link to when you press the button and display the loading screen, as for this, it might be as simple as displaying the icon before actually running the spawn code, and then removing it like 1 second later (as it will always be there while the game is frozen even if its already been removed serverside)

**Comment by FuzzyLeo** [comment]
> > Jazztronauts does this so it would be worth looking into how they do it
> 
> It's likely they just link to when you press the button and display the loading screen, as for this, it might be as simple as displaying the icon before actually running the spawn code, and then removing it like 1 second later (as it will always be there while the game is frozen even if its already been removed serverside)

i don't believe this is true as i've had it freeze at other points while playing the game and the same thing plays, theirs is an overall "the game is frozen to play this thing"

**Comment by AmyJeanes** [comment]
The location finding already runs as a coroutine (effectively in the background) and displays it's status with a message in the chat, "Finding location for interior" etc, I am not sure if you're referring to this or more the full game lock up when spawning in a new TARDIS that hasn't had it's models pre-loaded on the client yet which freezes the game for some time.

To fix this, depending on how `util.PrecacheModel` works you may also be able to run this sort of in the background in a similar fashion to the finding position code before attempting to actually spawn anything. It will still freeze up slightly but it will be mini freezes instead of one big freeze and we could display some progress bars / spinner.
